### PR TITLE
Add support for strict Content Security Policy

### PR DESCRIPTION
**Pull request by Dosenpfand**
To support the usage of Boostrap-Flask together with a [strict Content Security Policy (CSP)](https://csp.withgoogle.com/docs/strict-csp.html), to have a defense in depth against XSS attacks, the following changes have been made:

1. Remove inline style declarations, use classes instead
2. Remove inline JS, use HTML + CSS instead
3. Add support of nonces with JS includes
4. Add basic usage documentation

If desired I can still add an example and/or test case for this use case.

### CODE REVIEW

File: flask_bootstrap/templates/base/table.html [hunk 1]
```diff
@@ -154,11 +154,9 @@
...
                   {% endif %}
                 <input type="hidden" name="csrf_token" value="{{ csrf_token() }}"/>
-                <a class="action-icon text-decoration-none"
-                    href="javascript:{}"
-                    onclick="this.closest('form').submit();return false;"
-                    title="{{ config['BOOTSTRAP_TABLE_DELETE_TITLE'] }}">
+                <button type="submit" class="btn btn-link p-0 border-0" title="{{ config['BOOTSTRAP_TABLE_DELETE_TITLE'] }}">
                     {{ render_icon('trash-fill') }}
+                  </button>
                 </a>
```
Comment (greyli) [inline flask_bootstrap/templates/base/table.html:160]: The closing `a` tag should be removed.

@PanderMusubi Can you help to verify if the button change here is working as expected?

**Review by greyli — CHANGES_REQUESTED** [review]

**Comment by greyli** [comment]
Thanks for working on this. Please add tests and update the changelog.

File: flask_bootstrap/templates/bootstrap5/form.html [hunk 2]
```diff
@@ -6,7 +6,7 @@
         {%- for fieldname, errors in form.errors.items() %}
             {%- if bootstrap_is_hidden_field(form[fieldname]) %}
                 {%- for error in errors %}
-                    <div class="invalid-feedback" style="display: block;">{{ error }}</div>
```
[the remark this answers is not in the document]
Comment (Dosenpfand) [reply flask_bootstrap/templates/bootstrap5/form.html:9]: Reverted.

File: flask_bootstrap/templates/base/table.html (hunk 1 above)
Comment (Dosenpfand) [reply flask_bootstrap/templates/base/table.html]: Removed.

**Comment by Dosenpfand** [comment]
Thanks for the feedback. I added the changelog entry, added tests and fixed the 2 findings.

File: flask_bootstrap/templates/base/table.html (hunk 1 above)
Comment (Dosenpfand) [reply flask_bootstrap/templates/base/table.html]: They should be on the same height now.

Comment (greyli) [reply flask_bootstrap/templates/base/table.html]: Good job!

**Review by greyli — APPROVED** [review]

**Comment by greyli** [comment]
Merged, thanks!

**Pull request merged**